add option max_reflections_per_experiment to combine_experiments

dwpaley opened the pull request:
Analogous to min_reflections_per_experiment. Experiments are thrown out if they have more than the maximum reflections. This is meant to reject obvious multicrystal shots from xfel experiments, when combining expts after spotfinding.

graeme-winter commented on command_line/combine_experiments.py:
```diff
@@ -523,7 +530,13 @@ def average_detectors(target, panelgroups, depth):
                 ):
-                    skipped_expts += 1
+                    skipped_expts_min_refl += 1
+                    continue
```
Code coverage report suggests that this is not tested currently - could be worth adding a test? There's probably plenty of useful example data in `dials_data` which could be used for this.

dwpaley commented:
Thanks, Graeme. The application on my mind is for small-molecule still
shots where the biggest possible number of spots in a (single-crystalline)
frame is much smaller. So yes there's a range but if I see 100 spots I know
it can be rejected.

I added a test for min_refl and max_refl and will go ahead and merge.

Dan

On Thu, Aug 6, 2020 at 12:16 AM Graeme Winter <notifications@github.com>
wrote:

> *@graeme-winter* approved this pull request.
>
> Not convinced by the use case for this *however* that is orthogonal to
> the review process 🙂
>
> Why do I say this? Because the number of reflections you get *is*
> dependent on the number of lattices but is *also* dependent on things
> like the sample mosaicity and diffracting resolution. That said, *caveat
> utilitor*, if that's what they want then 🤷‍♂️ OK then.
>
> Suggest adding a test for this (which could test min_refl option at the
> same time, perhaps, and be a net improvement) as well as the obligatory
> news fragment. Added code looks to be in keeping with the surrounding area
> 👍
> ------------------------------
>
> In command_line/combine_experiments.py
> <https://github.com/dials/dials/pull/1369#discussion_r466192808>:
>
> > @@ -523,7 +530,13 @@ def average_detectors(target, panelgroups, depth):
>
>                      params.output.min_reflections_per_experiment is not None
>
>                      and n_sub_ref < params.output.min_reflections_per_experiment
>
>                  ):
>
> -                    skipped_expts += 1
>
> +                    skipped_expts_min_refl += 1
>
> +                    continue
>
>
> Code coverage report suggests that this is not tested currently - could be
> worth adding a test? There's probably plenty of useful example data in
> dials_data which could be used for this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dials/dials/pull/1369#pullrequestreview-462243765>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFLP5HYG4MTCKQHWG7BPSNDR7JKFNANCNFSM4PWBYBCQ>
> .
>

dwpaley commented:
Oops, I rebased this locally and pushed to master instead of merging this PR. @ndevenish please feel free to correct the history if needed. Let me know if you need anything from me. Thanks, Dan

ndevenish commented:
Hi Daniel,

Since there was nothing more recent and noticed relatively quickly I've rolled it back so we can properly merge this. 

As Graeme alluded to, please add a `newsfragments/1369.feature` text file with a one-line description of the changes (this goes into the CHANGELOG). It looks like you should probably add yourself to the `AUTHORS` file also.

After that, feel free to merge the pull request (make sure to squash merge this!).

Thanks!

dwpaley commented:
Thanks for the help! Sorry to cause trouble. I added the newsfragment and my name and squash merged this PR.
Best,
Dan
